### ISSUE TITLE
'make check' fails for a debug build

### ISSUE THREAD

**Issue by mhoes** [Fish-Git/hyperion#8]
Hi,

I get the following error when running 'make check' on Linux for a 'debug build':

```
$ configure --enable-debug --enable-getoptwrapper --enable-ipv6 --enable-cckd-bzip2 --enable-het-bzip2 --enable-object-rexx --enable-regina-rexx --enable-interlocked-access-facility-2=yes --enable-optimization=no

$ make check
Making check in m4
make[1]: Entering directory '/home/maarten/src/fish-git-hyperion-topdir/build/m4'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/maarten/src/fish-git-hyperion-topdir/build/m4'
Making check in util
make[1]: Entering directory '/home/maarten/src/fish-git-hyperion-topdir/build/util'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/maarten/src/fish-git-hyperion-topdir/build/util'
Making check in html
make[1]: Entering directory '/home/maarten/src/fish-git-hyperion-topdir/build/html'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/maarten/src/fish-git-hyperion-topdir/build/html'
Making check in man
make[1]: Entering directory '/home/maarten/src/fish-git-hyperion-topdir/build/man'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/maarten/src/fish-git-hyperion-topdir/build/man'
Making check in .
make[1]: Entering directory '/home/maarten/src/fish-git-hyperion-topdir/build'
make[1]: Leaving directory '/home/maarten/src/fish-git-hyperion-topdir/build'
../hyperion/tests/runtest -p .libs ../hyperion/tests
Files: ../hyperion/tests/agf.tst ../hyperion/tests/awsbsf.tst ../hyperion/tests/bfp-001-divtoint.tst ../hyperion/tests/bfp-002-loadr.tst ../hyperion/tests/bfp-003-loadfpi.tst ../hyperion/tests/bfp-004-cvttolog.tst ../hyperion/tests/bfp-005-cvttolog64.tst ../hyperion/tests/bfp-006-cvttofix.tst ../hyperion/tests/bfp-007-cvttofix64.tst ../hyperion/tests/bfp-008-cvtfrlog.tst ../hyperion/tests/bfp-009-cvtfrlog64.tst ../hyperion/tests/bfp-010-cvtfrfix.tst ../hyperion/tests/bfp-011-cvtfrfix64.tst ../hyperion/tests/bfp-012-loadtest.tst ../hyperion/tests/bfp-013-comps.tst ../hyperion/tests/bfp-014-divide.tst ../hyperion/tests/bfp-015-sqrt.tst ../hyperion/tests/bfp-016-add.tst ../hyperion/tests/bfp-017-loadl.tst ../hyperion/tests/bfp-018-subtract.tst ../hyperion/tests/bfp-019-multiply.tst ../hyperion/tests/bfp-020-multlonger.tst ../hyperion/tests/bfp-021-multadd.tst ../hyperion/tests/bfp-022-multsub.tst ../hyperion/tests/cipher.tst ../hyperion/tests/cmd-abs.tst ../hyperion/tests/cmd-rv.tst ../hyperion/tests/csxtr.tst ../hyperion/tests/digest.tst ../hyperion/tests/hetbsf.tst ../hyperion/tests/ilc.tst ../hyperion/tests/invpsw.tst ../hyperion/tests/kimd-hw.tst ../hyperion/tests/klmd-hw.tst ../hyperion/tests/kmac-hw.tst ../hyperion/tests/kmc-hw.tst ../hyperion/tests/kmctr-hw.tst ../hyperion/tests/kmf-hw.tst ../hyperion/tests/km-hw.tst ../hyperion/tests/kmo-hw.tst ../hyperion/tests/leapfrog.tst ../hyperion/tests/logicimm.tst ../hyperion/tests/mainsize.tst ../hyperion/tests/mhi.tst ../hyperion/tests/mvcle.tst ../hyperion/tests/pfpo.tst ../hyperion/tests/privop.tst ../hyperion/tests/problem.tst ../hyperion/tests/pr.tst ../hyperion/tests/runtest0.tst ../hyperion/tests/runtest4.tst ../hyperion/tests/semipriv.tst ../hyperion/tests/sigp.tst ../hyperion/tests/ssk370.tst ../hyperion/tests/sske370.tst ../hyperion/tests/sske390.tst ../hyperion/tests/sske.tst ../hyperion/tests/stfl.tst ../hyperion/tests/timeout.tst ../hyperion/tests/wild.tst ../hyperion/tests/zeos.tst
Variable ptrsize set to "8".
Variable platform set to "Linux".
Not variable assignment in argument string:  ../hyperion/tests
Novalue in /home/maarten/src/fish-git-hyperion-topdir/hyperion/tests/redtest.rexx   -- variable OKS
   738 >>>    oks = oks + 1
  This is often caused by missing or misspelled *Testcase
  Note that the verbs are case sensitive.  E.g., *testcase is not correct.
   767 +++                Signal Value lets_get_a_traceback
   563 +++             Call test lastmsg.ix = rest, msgtype 'message mismatch. ' info , 'Want:' rest, 'Got: ' lastmsg.ix
   348 +++          call msg 'Error'
   266 +++       Call order
    97 +++    Call procline l
Error 16 running "/home/maarten/src/fish-git-hyperion-topdir/hyperion/tests/redtest.rexx", line 767: Label not found
Error 16.1: Label "LETS_GET_A_TRACEBACK" not found
Makefile:2839: recipe for target 'check' failed
make: *** [check] Error 240
```

**Comment by Fish-Git** [comment]
I will try to reproduce it and get it fixed.

Thank you for reporting it.

**QUESTION**:

Does this happen _**only**_ with Debug builds?

Or does it also happen with normal Release builds too?

Thanks.

**Comment by Fish-Git** [comment]
At the very end of `Makefile.am` I am seeing:

    check:
    if APPLE
	    $(top_srcdir)/tests/runtest          $(top_srcdir)/tests
    else
	    $(top_srcdir)/tests/runtest -p .libs $(top_srcdir)/tests
    endif

I suspect the `-p .libs` should not be there.

When I manually run `/tests/runtest  ./tests  >  ~/runtest.log  2>&1`, all tests pass successfully:

    [...]
    Test sske#2.  53 OK compares.  All pass.
    Test stfl and stfle.  12 OK compares.  All pass.
    Test timeout.  1 OK compares.  All pass.
    Test multix.  2 OK compares.  All pass.
    Test multix1.  2 OK compares.  All pass.
    Test zeos.  6 OK compares.  All pass.
    Done 277 tests.  All OK.

I don't know why John felt `-p .libs` should be there. It seems dubious to me. I will try to reproduce your problem both with and without that parameter to see if that fixes your problem.

You can try it for yourself too in the mean time: simply remove the `if APPLE`, `else` and `endif` statements (as well as the suspect runtest statement) leaving ONLY the plain runtest command without and extra arguments. Then run your configure and make and make check again.

**Comment by Fish-Git** [comment]
Okay, I have tested `make check` with my Fish-Git Hyperion for both a normal Release mode build _as well as_ for a `--enable-debug` Debug, and both work just fine as-is (i.e. with the existing `-p .libs` argument for the non-APPLE case).

In other words, I am unable to reproduce your problem. &nbsp; `:(`

I will try the same tests again without the `-p .libs` argument to verify that works okay or not as well.

_**HOWEVER**_, I happened to notice that it _appears_, in your original problem report, that you simply did a `configure` followed immediately by a `make check`. If this is true, then this is very likely your problem! You must run `make` _before_ you can do a `make check`!!

If you didn't do that, then please try again, e.g.:

    configure ... && make && make check

Let me know if that works okay for you.

Thanks!

**Comment by Fish-Git** [comment]
Okay, I have completed the full sweet of tests for both in-source builds as well as for out-of-source builds for both Debug builds as well as normal Release builds, both using runtest with the extra (but unneeded) -p .libs argument as well as without the -p .libs argument, and everything works beautifully. I an unable to reproduce your failure.

I strongly suspect (but did not bother to confirm) that you simply failed to run `make` before your `make check`.  Please ensure you always to a `make` (after your `configure`) _before_ you run `make check`.

I am closing this issue as I believe it to be a user error and am unable to reproduce the failure.

**Comment by mhoes** [comment]
Hi,

I did run 'make' to build Hyperion before I run the 'make check'. It only happens for a debug build, and not for a regular build. Leaving the "-p .libs" out of the command, and just running '../hyperion/tests/runtest ../hyperion/tests" makes no difference and results in the same error. I'm doing an 'out of source build'. I have attached my 'allTests.out', perhaps something can be learned from that.


[allTests.out.gz](https://github.com/Fish-Git/hyperion/files/784938/allTests.out.gz)

**Comment by Fish-Git** [comment]
Maarten,

I took a closer look at the tests I ran and apparently I wasn't testing what I thought I was testing!

You are 100% correct: `make check` does indeed fail _exactly_ as you described for out-of-source Debug builds, and thanks to the `allTests.out` file you provided, I now know why.

My sincerest apologies for dismissing your problem too quickly!

Stay tuned. I shall have a fix for you shortly.

Thank you for your patience and persistence.

**Comment by Fish-Git** [comment]
Should be fixed now by commit eec676d5fb3b50d5e9df369177dc8dca86f6c1ce.

Please re-pull and rebuild to confirm (`autogen.sh`, `configure ...`, `make`, `make check`).

I will await your confirmation first before closing this issue.

Thanks.

**Comment by mhoes** [comment]
Thanks, that commit does indeed fix the issue for me.
